src/io/fileops.c

```diff
@@ -229,13 +229,15 @@ MVMint64 MVM_file_isexecutable(MVMThreadContext *tc, MVMString *filename, MVMint
                     MVMString *fileext = MVM_string_substring(tc, filename, n, -1);
                     char *ext  = MVM_string_utf8_c8_encode_C_string(tc, fileext);
                     char *pext = getenv("PATHEXT");
-                    int plen   = strlen(pext);
-                    int i;
-                    for (i = 0; i < plen; i++) {
-                        if (0 == stricmp(ext, pext++)) {
-                             r = 1;
-                             break;
+                    char *delim = ";";
+                    char *ptr = strtok(pext, delim);
```

Comment:
From the man page of strtok:
Be cautious when using these functions.  If you do use them, note that:
* These functions modify their first argument.
* These functions cannot be used on constant strings.
* The identity of the delimiting byte is lost.
* The strtok() function uses a static buffer while parsing, so it's not thread safe.  Use strtok_r() if this matters to you.

All of these except for the bit about constant strings are real issues. E.g. this file test will only work once and then only find matches for the very first extension in PATHEXT. There's also gonna be issues if any other thread uses strtok for any purpose concurrently (which may happen with poorly written native libraries which are not aware that they are used in a multi threaded program)